### ISSUE TITLE
support filtering by DONE

### ISSUE THREAD

**Issue by lcb931023** [dotcli/memex#3]
support filtering to only see items that aren't done yet - items that I haven't finished consuming.

Possible Font Awesome icon to use:

[spinner](https://fontawesome.com/icons/spinner?style=solid)

[tasks](https://fontawesome.com/icons/tasks?style=solid)

**Comment by kormyen** [comment]
Nice :D

I was planning to have something like:

**_tick icon_**   33%
**_cross icon_** 67%

**Comment by kormyen** [comment]
https://github.com/kormyen/memex/commit/baf4b2f8b7c8357a737947920a952af6506fe8fe

**Comment by lcb931023** [comment]
wow, awesome!

I'll have to do a huge pull from your repo soon, to get these new functionalities.

Speaking of tracking progress of consuming memex bookmarks: how often do you read through or reference items in memex? For most times interacting with memex, I find myself adding or adjusting items, instead of retrieving memories from it. Feels a bit like hoarding.

**Comment by kormyen** [comment]
Cool. Still got to finish up the 'add' functionality to a reasonable standard, just been busy with other things lately - mostly https://github.com/kormyen/vns and a kitchen bench.

Same regarding hoarding. For me it is not necessarily a bad thing - kind of the main point to be an external memory storage location that is quick to narrow down (filter by xyz then search) to find a thing when needed. I'm also using it to deliberately collect word terms and quotes.

I've been thinking at the back of my mind of a simple-ish system for memex to show you x (like 10 or 5) items/actions it wants you to action each day, intending for it to become a routine.
For example: maybe it is configured to show you the 5 worst entries (in terms of least amount of information/tags/missing type etc) expecting you to fill them in or delete them. Taking inspiration from Devine's https://wiki.xxiivv.com/#tracker

Or entries could have an optional field of reminding you about that entry every z days (taking inspiration from Anki http://augmentingcognition.com/ltm.html ) . Or use that 'revisit' field, maybe app settings to specify how often it should show you a 'REVI = true' entry again (eg: 365 days, but max 2 in one day)

I was also pondering if porting it to run on https://datproject.org/ / Beaker would make sense. Solving the web-accessibility with auto-backing up while also being editable in the browser with Dat functionality still without a server as such. Haven't researched it though.

Most of all I'd like to improve actual note taking / note editing / note collection / note overview. Notes are a problem for me because I'll often write down a lot of thoughts on a thing/project and then either loose them, or just not revisit them.

Maybe a way to encourage culling least important entries (or no longer needed) wouldn't be a bad idea.

It'll need progressive loading of entries before too long.

This kind of seems like a long term project for me, I'll chip away at features every now and again.

100% open to your thoughts and opinions. No pressure.

**Comment by lcb931023** [comment]
Wow- this is excellent. I've been only using this as a fancy bookmark tool, but if the goal is "memory extender" then the priorities would be different - emphasizing terms and quotes make sense then.

In that case it's actually quite different from 'bookmark' tool, which rotates around links. One potential issue is the "lists" and "wiki" items in here - they would be very difficult to revisit and get "done" the same way a word or quote gets memorized, since they are portals to whole bunch of content.

Thanks for the write-up. I'm reading up on the augmentingcognition.com article. Will have more thoughts to share soon...

**Comment by kormyen** [comment]
Yeah. I'm personally not interested in memorizing facts Anki-style but I love the concepts in that article and the tools.
To me memex is more just a notebook / external memory / augmentation (like a mobile phone / internet) and the goal is to just keep adding to it and making it smarter / more helpful.

Really though to me its just stepping stones to building my website :D

I think revising those 'lists' and 'wiki' items is where that Anki style thing could work though?
Example: I really like (wish to revisit) https://wiki.nikitavoloboev.xyz . Would be nice if every two months (customizable) memex would remind me to visit it to see what is new.
eg: Visit Knowledge Wiki [tomorrow] [done (repeat)] [extend (anki extend reminder delay)]

Reminds me: there was a website where each person would share the/one website that they visit once a day, the/one website that they visit once a week, the/one website that they visit once a month, and the/one website that they visit once a year.
